### PR TITLE
Move utility functions into own helper class

### PR DESCRIPTION
**Pull request by VSEphpbb**
There's not really any programing logic changed here. instead this PR is moving some sub-task methods out of the base class and into its own helper class.

### CODE REVIEW

**Comment by rxu** [comment]
The PR looks good to me, although I'd think about possible moving more functions into the helper, f.e. ```get_group_rules()```.

File: conditions/manager.php [hunk 1]
```diff
@@ -313,4 +313,16 @@ public function sync_autogroups($autogroups_rule_id)
...
+	/**
+	* Helper to load objects (autogroup types)
+	*
+	* @param string     $type_name      The name of the auto group type
+	*
+	* @return object The auto group type service
+	*/
+	protected function get_type($type_name)
+	{
+		return $this->phpbb_container->get($type_name);
```
Comment (michaelcullum) [inline conditions/manager.php]: This is really un-necessary...

File: conditions/type/base.php [hunk 2]
```diff
@@ -68,6 +66,10 @@ public function __construct(ContainerInterface $container, \phpbb\config\config
 		$this->phpbb_root_path = $phpbb_root_path;
 		$this->php_ext = $php_ext;
 
+		// Load the helper from the container instead of dependency injection to
+		// help anyone extending auto groups avoid setting optional dependencies.
```
Comment (michaelcullum) [inline conditions/type/base.php]: hm?

Comment (VSEphpbb) [inline conditions/type/base.php]: If somebody writes an extension that also extends autogroups for their own purposes, and we do this as a service, they will have to have the $helper treated as an optional dependency, which is ugly.

File: conditions/manager.php (hunk 1 above)
Comment (VSEphpbb) [inline conditions/manager.php]: I was thinking we might be able to use this in other classes instead of always passing in the container.

**Comment by VSEphpbb** [comment]
Reverted the useless-ness

**Comment by VSEphpbb** [comment]
@michaelcullum bump

**Pull request merged**